src/std/misc/string.ss

```diff
@@ -185,3 +187,30 @@
 ;;  (string-whitespace? " \n\r \t") => #t
 (def (string-whitespace? s)
   (string-every char-whitespace? s))
+
+
+(def (random-word-char)
+  (declare (not safe) (fixnum))
+  (def n (random-integer 63))
+  (integer->char
+   (+ n (cond
+	 ((< n 10) 48) ; 0-9
+	 ((< n 36) 55) ; A-Z
+	 ((< n 62) 61) ; a-z
+	 (else 33))))) ; _
+
+
+;; random-string returns a string consisting of
+;; regex word-boundary characters [a-zA-Z0-9_].
+;;
+;; Example:
+;;  (random-string) => "5CfMyYd2Ob"
+(def (random-string (len 10))
+  (declare (not safe) (fixnum))
+  (if (> len 0)
```

Comment:
we need to add a check for len being a fixnum here and throw an error if it is not.